tests/template_tests/test_logging.py

```diff
@@ -75,7 +75,28 @@ def test_log_on_variable_does_not_exist_not_silent(self):
         raised_exception = self.test_handler.log_record.exc_info[1]
         self.assertEqual(
             str(raised_exception),
-            'Failed lookup for key [author] in %r' % ("{%r: %r}" % ('section', 'News'))
+            'Failed lookup for key [author] in %r' % ({'section': 'News'},)
+        )
+
+    def test_log_on_variable_does_not_exist_does_not_stringify(self):
+        class TestObject:
+            def __str__(self):
+                return 'fail'
+
+        test_object = TestObject()
+
+        with self.assertRaises(VariableDoesNotExist):
```

Comment:
or 
```
msg = 'Failed lookup for key [article] in %r' % test_object
with self.assertRaisesMessage(VariableDoesNotExist, msg):
```
Actually, I wonder if this test is a bit more complicated than needed. Would it be enough to test `VariableDoesNotExist.__str__()` directly?